docs/cvl/methods.md

```diff
@@ -178,6 +190,63 @@ shown in the web report can indicate whether a summary was applied.
+   Example: `C.foo(uint)`
+2. Wildcard contract - a pattern specifying the function signature to match this signature on the primary contract and all linked contracts.
+   Example: `_.bar(address)`
+3. Wildcard function - a pattern specifying the primary or a linked contract, and matches all external functions in specified contract.
+For the default summary the user can choose one of: `HAVOC_ALL`, `HAVOC_ECF`, `NONDET`.
+
+To better understand this functionality an example entry is useful.
+The following entry will specify three functions to route calls to:
+1. `C.foo(uint)`
+2. `Other.bar(address)`
+3. `C.baz(bool)`
+```cvl
+methods {
+   function _._ external => DISPATCH [
+      C.foo(uint),
+      _.bar(address), // Will resolve to all linked function with the signature "bar(address)", specifically Other.bar(address)
+      C._ // Will resolve to all functions in C, specifically C.foo(uint) and C.baz(bool)
+   ] default NONDET;
+}
+```
+Entry flags (`envfree`, `returns`, and `optional`) are not allowed on an unresolved-calls entry. 
+Also, the visibility is always external, and no policy should be specified.
+
+For an unresolved function call being summarized with the dispatch list above, the Prover will replace the call with a dynamic resolution of the function call. That is something in the lines of:
+```solidity
+function summarized(address a, bytes calldata data) external {
+  if (uint32(data[0:4]) == 0x11111111 && address == address(c)) {
+    // Function selector was equal to foo's
+    // Call C.foo(...)
+  } else if (uint32(data[0:4]) == 0x22222222 && address == address(o)) {
+    // Function selector was equal to bar's
+    // Call O.bar(...)
+  } else if (uint32(data[0:4]) == 0x33333333 && address == address(c)) {
+    // Function selector was equal to baz's
+    // Call C.baz(...)
+  } else {
+    // Use the default summary which is, in this case, NONDET.
+  }
+}
+```
+
+Dynamic resolution in Solidity refers to the process of determining the specific function to call at runtime based on the function signature and the target contract address. In the provided example, when an unresolved function call is encountered, the Prover dynamically resolves it by inspecting the function selector in the transaction data and the target contract address. By comparing the function selector against known signatures and verifying the contract address, the Prover identifies the appropriate function to call. This dynamic resolution mechanism is crucial for refining specifications because it enables the Prover to accurately model the behavior of smart contracts, even when the exact function being called is not known statically. By replacing unresolved calls with dynamically resolved calls in the dispatch list summary, the specification becomes more precise, leading to more accurate verification results and improved assurance in the correctness of the smart contract.
```

Comment:
But it is not in Solidity, the above is just pseudo code explaining what dispatch list is doing

Reply:
Wow that's a big miss on my part. I'll keep the paragraph but filter through it so it will be clear that this describes the resulting code